src/lib/synergy/ProtocolUtil.cpp

```diff
@@ -29,6 +30,57 @@
 // ProtocolUtil
 //
 
+namespace  {
+
+void
+writeInt(UInt32 Value, UInt32 Length, std::vector<UInt8>& Buffer)
+{
+    switch(Length)
+    {
+        case 1:
+            Buffer.push_back(static_cast<UInt8>(Value & 0xff));
+            break;
+        case 4:
+            Buffer.push_back(static_cast<UInt8>((Value >> 24) & 0xff));
+            Buffer.push_back(static_cast<UInt8>((Value >> 16) & 0xff));
+            Buffer.push_back(static_cast<UInt8>((Value >>  8) & 0xff));
+            Buffer.push_back(static_cast<UInt8>( Value        & 0xff));
+            break;
+        case 2:
+            Buffer.push_back(static_cast<UInt8>((Value >> 8) & 0xff));
+            Buffer.push_back(static_cast<UInt8>( Value       & 0xff));
```

Comment:
Looking im my IDE a Clang tidy error happens here. 
> Use of a signed integer operand with a binary bitwise operator
Wondering if we should update the numbers and hex to specifically be unsigned for example 
```suggestion
            Buffer.push_back(static_cast<UInt8>(Value & 0xffu));
            break;
        case 4:
            Buffer.push_back(static_cast<UInt8>((Value >> 24u) & 0xffu));
            Buffer.push_back(static_cast<UInt8>((Value >> 16u) & 0xffu));
            Buffer.push_back(static_cast<UInt8>((Value >>  8u) & 0xffu));
            Buffer.push_back(static_cast<UInt8>( Value        & 0xffu));
            break;
        case 2:
            Buffer.push_back(static_cast<UInt8>((Value >> 8u) & 0xffu));
            Buffer.push_back(static_cast<UInt8>( Value       & 0xffu));
```
As described here https://stackoverflow.com/a/50399245/11680140